src/kernel/include/thd.h

```diff
@@ -493,34 +532,56 @@ thd_current_pgtbl(struct thread *thd)
 	/* don't use the cached invstk_top here. We need the stack
 	 * pointer of the specified thread. */
 	curr_entry = &thd->invstk[thd->invstk_top];
-	return curr_entry->comp_info.pgtbl;
+	return curr_entry->comp_invstk_info.comp_info.pgtbl;
```

Comment:
blank line before this plz. I know the style used here isn't part of your change but you might as well fix it.